Feature/#2: Minimum required styles for accordion panels when transitions applied

smillart opened the pull request:
This pull request contains default styles for accordion panels where **CSS transitions** can be applied from hidden to revealed state (and vice versa).

Commit:
Updated Accordion minimum required styles by preparing transition effect (option) for accordion panels

Commit:
Added animation transition with pure css

smillart commented on dist/accordion/css/accordion.min.css:
```diff
@@ -1 +1,62 @@
+
+/* For accordion panels with a transition effect */
+.aria-accordion__panel--transition {
+  display: block;
```
```suggestion
```

`display: block;` is unnecessary at this point since the script checks if the DOM is correct and matches with the required `<div data-aria-accordion-panel>` (block-level) element... as expected.

smillart commented on dist/accordion/css/accordion.min.css:
```diff
@@ -1 +1,62 @@
+/* For accordion panels with a transition effect */
+.aria-accordion__panel--transition {
+  display: block;
+  max-height: 100vh;
```
```suggestion
  max-height: 80vh;
```

I tested with `80vh` as value and it ensures the visibility for both **Accordion Header** and its associated **Accordion Panel** inside the available viewport in case of very very long content.

smillart commented on dist/accordion/css/accordion.min.css:
```diff
@@ -1 +1,62 @@
+.aria-accordion__panel--transition {
+  display: block;
+  max-height: 100vh;
+  visibility: visible;
```
```suggestion
  visibility: visible;
  opacity: 1;
  height: auto;
```

Added `opacity` CSS property for a gradual appearance of the panel content.
Added `height` CSS property to ensure a default height when the panel opened.

smillart commented on dist/accordion/css/accordion.min.css:
```diff
@@ -1 +1,62 @@
+  display: block;
+  max-height: 100vh;
+  visibility: visible;
+  -webkit-transition: max-height 800ms ease-in;
+  -moz-transition: max-height 800ms ease-in;
+  -ms-transition: max-height 800ms ease-in;
+  -o-transition: max-height 800ms ease-in;
+  transition: max-height 800ms ease-in;
```
```suggestion
  -webkit-transition:
    max-height 0.3s,
    padding-top 0.3s,
    padding-bottom 0.3s,
    opacity 0.5s;
  -o-transition:
    max-height 0.3s,
    padding-top 0.3s,
    padding-bottom 0.3s,
    opacity 0.5s;
  transition:
    max-height 0.3s,
    padding-top 0.3s,
    padding-bottom 0.3s,
    opacity 0.5s;
```
Last 3 browser versions is enough (IE9+10+11) regarding vendor prefixes. Also, we need `padding` and `opacity` transitions. No value specifies the default `ease` one for the speed curve ensuring a transition effect with a slow start, then fast, then end slowly.

smillart commented on dist/accordion/css/accordion.min.css:
```diff
@@ -1 +1,62 @@
+  -ms-transition: max-height 800ms ease-in;
+  -o-transition: max-height 800ms ease-in;
+  transition: max-height 800ms ease-in;
+  overflow: visible;
```
```suggestion
  overflow: auto;
```
We need a scrollbar in case of very very long panel content.

smillart commented on dist/accordion/css/accordion.min.css:
```diff
@@ -1 +1,62 @@
+
+.aria-accordion__panel--transition[hidden] {
+  display: block;
+  height: auto;
```
```suggestion
```
Moved to the previous `.aria-accordion__panel--transition` selector to ensure a default height when the panel opened.

smillart commented on dist/accordion/css/accordion.min.css:
```diff
@@ -1 +1,62 @@
+  padding-top:0;
+  padding-bottom: 0;
+  overflow: hidden;
+  -webkit-transition: max-height 800ms ease-out;
+  -moz-transition: max-height 800ms ease-out;
+  -ms-transition: max-height 800ms ease-out;
+  -o-transition: max-height 800ms ease-out;
+  transition: max-height 800ms ease-out;
```
```suggestion
```

Transitions in the previous `.aria-accordion__panel--transition` selector only is sufficient and ensure transition effect on both hidden and revealed states.

smillart commented on dist/accordion/css/accordion.min.css:
```diff
@@ -1 +1,62 @@
+  -ms-transition: max-height 800ms ease-out;
+  -o-transition: max-height 800ms ease-out;
+  transition: max-height 800ms ease-out;
+  visibility: visible;
```
```suggestion
  visibility: hidden;
  opacity: 0;
```
We need `hidden` panel content to avoid screen readers reading it.
Added `opacity` CSS property for a gradual disappearance of the panel content.

smillart commented on dist/accordion/css/accordion.min.css:
```diff
@@ -1 +1,62 @@
+  -o-transition: max-height 800ms ease-in;
+  transition: max-height 800ms ease-in;
+  overflow: visible;
+  /* persist overflow value from animation */
+  animation: 800ms delay-overflow;
+  animation: 900ms delay-padding-expanding;
```
```suggestion
```

Regarding `animation` CSS properties, it is very cool and powerful. However, I think that here (in the library context) as minimum required styles for transition we do not need it. You can use it for the CKEditor plug-in or on a per-project basis as additional styles.

smillart commented on dist/accordion/css/accordion.min.css:
```diff
@@ -1 +1,62 @@
+  -o-transition: max-height 800ms ease-out;
+  transition: max-height 800ms ease-out;
+  visibility: visible;
+  /* persist visibility value from animation */
+  animation: 800ms delay-visibility;
+  animation: 900ms delay-padding-collapse;
+}
+
+@keyframes delay-visibility {
+  from {
+    visibility: visible;
+  }
+}
+
+@keyframes delay-overflow {
+  from {
+    overflow: hidden;
+  }
+}
+@keyframes delay-padding-collapse {
+  from {
+    padding-top:1rem;
+    padding-bottom:1rem;
+  }
+}
+@keyframes delay-padding-expanding {
+  from {
+    padding-top:0;
+    padding-bottom:0;
+  }
+}
```
```suggestion
```
See my comment here above regarding `animation` CSS properties.

laurent-b4y reviewed (approved):
Seems good, thanks for considering my input. All of your comments make sense to me.

Commit:
Issue #2: Removed unnecessary `display: block;` since `<div data-aria-accordion-panel>` is a block-level element by default

`display: block;` is unnecessary at this point since the script checks if the DOM is correct and matches with the required `<div data-aria-accordion-panel>` (block-level) element... as expected.

Commit:
Issue #2: Changed `max-height` to `80vh` to ensure both Accordion Header/Pane visibility within the viewport

Commit:
Issue #2: Added `opacity` and `height` CSS properties for gradual appearance and default opened panel height

Commit:
Issue #2: Kept vendor prefixes for the last 3 browser versions adding `padding` and `opacity` transitions

Commit:
Issue #2: Changed `overflow` to `auto` to display a scrollbar in case of very very long panel content.

Commit:
Issue #2: Removed `animation` CSS properties to keep minimalist styles by default for transitions

Commit:
Issue #2: Moved to `.aria-accordion__panel--transition` rule to ensure default opened panel height

Commit:
Issue #2: Removed unnecessary and duplicated `transition` CSS properties

Commit:
Issue #2: Changed `visibility` to `hidden` to avoid screen readers reading it. Added `opacity` for panel disappearance

Commit:
Issue #2: Removed `animation` associated `@keyframes` to ensure minimalist styles by default for transitions

Commit:
Ignored all Sass source files (which need to stay on local host)

Commit:
Issue #2: Minified Accordion minimum required styles
